src/librustc_ast_lowering/path.rs

```diff
@@ -55,9 +55,6 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                         Res::Def(DefKind::AssocTy, def_id) if i + 2 == proj_start => {
                             Some(parent_def_id(self, def_id))
                         }
-                        Res::Def(DefKind::Variant, def_id) if i + 1 == proj_start => {
-                            Some(parent_def_id(self, def_id))
-                        }
```

Comment:
Why is this deleted?
This way we don't get the correct number of lifetime parameters in the `Enum` segment in `some::prefix::Enum::Variant`.

Reply:
The lifetimes are for elision, so the full path would need to be a type (e.g. in a function signature) for this to matter.

If we want to be future-compat with variant types, we can keep this and make it depend on which segment has generic args, like typeck does.

Reply:
Changed it in a backwards compatible way. Now the default behavior will remain the same _unless_ the `enum` path segment has arguments, and in that case _only that one_ will have the `Some(parent_def_id)`. This keeps the current behavior while also allowing the newer syntax to work correctly.